### PR TITLE
Some changes to make the Kyber impl more consistent with the stdlib

### PR DESCRIPTION
**Pull request by jedisct1**
- Variable names are snake_case
- Types names are PascalCase
- Make constants for lengths consistent with other similar functions, (ex: dh.X25519)
- PrivateKey -> SecretKey, also for consistency with the rest of the standard library (and to make the `pk` reduction less confusing)
- KeyPair.create() to create deterministic and randomized key pairs.

The number of bits in `usize` should be `Log2Int(usize)` to avoid portability issues with different `usize` sizes.

### CODE REVIEW

File: lib/std/crypto/kyber.zig [hunk 1]
```diff
@@ -228,13 +241,13 @@ fn Kyber(comptime p: Params) type {
                 kdf.squeeze(ss);
             }
 
-            pub fn pack(pk: PublicKey) [packedSize]u8 {
+            pub fn pack(pk: PublicKey) [packet_length]u8 {
```
Comment (bwesterb) [inline lib/std/crypto/kyber.zig]: packe**d**?

File: lib/std/crypto/kyber.zig [hunk 2]
```diff
@@ -985,55 +1000,55 @@ const Poly = struct {
     // Assumes p is normalized.
     fn compress(p: Poly, comptime d: u8) [compressedSize(d)]u8 {
         @setEvalBranchQuota(10000);
-        const Qover2: u32 = comptime @divTrunc(Q, 2); // (q-1)/2
-        const twoDm1: u32 = comptime (1 << d) - 1; // 2ᵈ-1
-        var inOff: usize = 0;
-        var outOff: usize = 0;
+        const q_over_2: u32 = comptime @divTrunc(Q, 2); // (q-1)/2
+        const two_dm_1: u32 = comptime (1 << d) - 1; // 2ᵈ-1
```
Comment (bwesterb) [inline lib/std/crypto/kyber.zig]: Bit weird now. `two_d_min_1` perhaps?

File: lib/std/crypto/kyber.zig (hunk 1 above)
Comment (jedisct1) [reply lib/std/crypto/kyber.zig]: Hahahaha obviously, yes :)

**Review by bwesterb — APPROVED** [review]

**Comment by bwesterb** [comment]
> /cc @bwesterb (not sure that you are notified about PRs in this repos)

Turned on notifications now. Bit weird it's not on by default.

**Pull request merged**